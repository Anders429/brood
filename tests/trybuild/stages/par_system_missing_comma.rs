use brood::{query::{filter, filter::Filter, result, views}, registry::ContainsParViews, system::{schedule::stages, ParSystem}};

struct MySystem;

impl<'a> ParSystem<'a> for MySystem {
    type Views = views!();
    type Filter = filter::None;

    fn run<R, FI, VI, P, I, Q>(&mut self, query_results: result::ParIter<'a, R, Self::Filter, FI, Self::Views, VI, P, I, Q>)
    where
        R: ContainsParViews<'a, Self::Views, P, I, Q> + 'a,
        Self::Filter: Filter<R, FI>,
        Self::Views: Filter<R, VI>, {}
}

type Stages = stages!{
    par_system: MySystem
    flush
};

fn main() {}
