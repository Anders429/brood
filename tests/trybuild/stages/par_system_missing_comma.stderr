error: no rules expected the token `flush`
  --> tests/trybuild/stages/par_system_missing_comma.rs:18:5
   |
18 |     flush
   |     ^^^^^ no rules expected this token in macro call
